src/ERC6538Registry.sol

```diff
@@ -24,22 +68,177 @@ contract ERC6538Registry is IERC6538Registry {
+  /// @dev The following code is from OpenZeppelin's `EIP712.sol` file.
+  /// @dev The method visibility was changed from internal to public. Original file permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/EIP712.sol
+  function _domainSeparatorV4() public view returns (bytes32) {
+    if (address(this) == _cachedThis && block.chainid == _cachedChainId) {
+      return _cachedDomainSeparator;
+    } else {
+      return _buildDomainSeparator();
+    }
   }
 
-  /// @dev Converts an `address` to `bytes`.
-  function _toBytes(address who) internal pure returns (bytes memory) {
-    return bytes.concat(bytes32(uint256(uint160(who))));
+  /// @notice Returns the hash of the EIP712 Domain Separator.
+  /// @dev The following code is from OpenZeppelin's `EIP712.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/EIP712.sol
+  function _buildDomainSeparator() private view returns (bytes32) {
+    return
+      keccak256(abi.encode(TYPE_HASH, _hashedName, _hashedVersion, block.chainid, address(this)));
   }
+
+  /// @notice Returns the hash of the fully encoded EIP712 message for this domain.
+  /// @dev The following code is from OpenZeppelin's `EIP712.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/EIP712.sol
+  /// @param structHash The hash of the struct containing the message data, as defined in
+  /// https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct
+  function _hashTypedDataV4(bytes32 structHash) internal view virtual returns (bytes32) {
+    return toTypedDataHash(_domainSeparatorV4(), structHash);
+  }
+
+  /// @dev Returns the keccak256 digest of an EIP-712 typed data (ERC-191 version `0x01`).
+  /// @dev The digest is calculated from a `domainSeparator` and a `structHash`, by prefixing them
+  /// with `\x19\x01` and hashing the result. It corresponds to the hash signed by the
+  /// https://eips.ethereum.org/EIPS/eip-712[`eth_signTypedData`] JSON-RPC method as part of
+  /// EIP-712.
+  /// @dev The following code is from OpenZeppelin's `MessageHashUtils.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/MessageHashUtils.sol
+  /// @param domainSeparator The domain separator returned by `_domainSeparatorV4`.
+  /// @param structHash The hash of the struct containing the message data, as defined in
+  /// https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct
+  function toTypedDataHash(bytes32 domainSeparator, bytes32 structHash)
+    internal
+    pure
+    returns (bytes32 digest)
+  {
+    /// @solidity memory-safe-assembly
+    assembly {
```

Comment:
same as above, though you may prefer to leave the comment-style to minimize the diff—again up to you

Reply:
Updated here in the same commit as above: https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/9372812c18bdef57e31ca84b7a0490e0d3089979